pkg/apply/task/delete_inv_task_test.go

```diff
@@ -4,65 +4,144 @@
 		},
-		"inventory not found is not error and not returned": {
-			err: apierrors.NewNotFound(schema.GroupResource{Resource: "simples"},
-				"unused-resource-name"),
-			isError: false,
+		"one failed delete; one inventory object": {
+			contextInventory: &actuation.Inventory{
+				TypeMeta:   invTypeMeta,
+				ObjectMeta: invObjMeta,
+				Spec:       actuation.InventorySpec{},
+				Status: actuation.InventoryStatus{
+					Objects: []actuation.ObjectStatus{
+						{
+							ObjectReference: ref1,
+							Strategy:        actuation.ActuationStrategyDelete,
+							Actuation:       actuation.ActuationFailed,
+							Reconcile:       actuation.ReconcileSkipped,
+						},
+					},
+				},
+			},
+			expectedInventory: &actuation.Inventory{
+				TypeMeta:   invTypeMeta,
+				ObjectMeta: invObjMeta,
+				Spec: actuation.InventorySpec{
+					Objects: []actuation.ObjectReference{ref1},
+				},
+				Status: actuation.InventoryStatus{
+					Objects: []actuation.ObjectStatus{
+						{
+							ObjectReference: ref1,
+							Strategy:        actuation.ActuationStrategyDelete,
+							Actuation:       actuation.ActuationFailed,
+							Reconcile:       actuation.ReconcileSkipped,
+						},
+					},
+				},
+			},
 		},
+		// TODO: make a FakeClient to test error handling
+		// "error is returned in result": {
+		// 	err:     apierrors.NewResourceExpired("unused message"),
+		// 	isError: true,
+		// },
+		// "inventory not found is not error and not returned": {
+		// 	err: apierrors.NewNotFound(schema.GroupResource{Resource: "simples"},
+		// 		"unused-resource-name"),
+		// 	isError: false,
+		// },
 	}
 	for name, tc := range testCases {
 		t.Run(name, func(t *testing.T) {
-			client := inventory.NewFakeClient(object.ObjMetadataSet{})
-			client.Err = tc.err
+			client := &inventory.InMemoryClient{}
+			// client.Err = tc.err
 			eventChannel := make(chan event.Event)
 			resourceCache := cache.NewResourceCacheMap()
-			context := taskrunner.NewTaskContext(eventChannel, resourceCache)
+			taskContext := taskrunner.NewTaskContext(eventChannel, resourceCache)
+
+			// initialize inventory in context
+			inv := taskContext.InventoryManager().Inventory()
+			tc.contextInventory.DeepCopyInto(inv)
 
 			task := DeleteInvTask{
 				TaskName:  taskName,
 				InvClient: client,
-				InvInfo:   localInv,
 				DryRun:    common.DryRunNone,
 			}
 			if taskName != task.Name() {
 				t.Errorf("expected task name (%s), got (%s)", taskName, task.Name())
 			}
-			task.Start(context)
-			result := <-context.TaskChannel()
-			if tc.isError {
-				if tc.err != result.Err {
-					t.Errorf("running DeleteInvTask expected error (%s), got (%s)", tc.err, result.Err)
-				}
-			} else {
-				if result.Err != nil {
-					t.Errorf("unexpected error running DeleteInvTask: %s", result.Err)
-				}
+			task.Start(taskContext)
+			result := <-taskContext.TaskChannel()
+			if tc.expectedError != nil {
+				assert.EqualError(t, result.Err, tc.expectedError.Error())
```

Comment:
It might be possible to pull out this assert.EqualError cleanup to its own PR. It can hopefully be done separately from the subsequent testutil.AssertEqual of the inventory after Load. But given the high amount of other test changes here, it's not really a big deal to pull out unless there are other similar changes in other test files.